pytorch_lightning/trainer/trainer.py

```diff
@@ -1168,6 +1171,18 @@ def call_hook(self, hook_name: str, *args, **kwargs) -> Any:
 
         return output
 
+    def _device_parsing(
+        self, gpus: Optional[Union[List[int], str, int]], auto_select_gpus: bool, tpu_cores: Optional[Union[List[int],
+                                                                                                            str, int]]
+    ) -> Tuple[Optional[List[int]], Optional[Union[List[int], int]]]:
+        if auto_select_gpus and isinstance(gpus, int):
+            gpus = pick_multiple_gpus(gpus)
+
+        # TODO (@seannaren, @kaushikb11): Add IPU parsing logic here
+        gpu_ids = device_parser.parse_gpu_ids(gpus)
```

Comment:
Let's have a warning on Trainer instantiation and actually raise an exception on Trainer.{fit, ...} call.

Reply:
Not needed, after the conversation with Thomas.